pkgs/by-name/si/simutrans/package.nix

```diff
@@ -2,196 +2,439 @@
+      # That is, we intentionally follow the XDG basedir spec, and **override the default behaviour of the executable here**,
+      # which, again, wouldn't normally follow the XDG basedir spec. It normally wants "${XDG_DATA_HOME:-$HOME}/simutrans" (which we override)
+      #
+      # Then, we use it to set defaults for SIMUTRANS_USERDIR and SIMUTRANS_INSTALLDIR.
+      # These should be '--set-default's, but wrapProgram will only accept literal strings here.
+      # We're reading from the environment, so we use --run instead.
+      #
+      # A user can overwrite these values:
+      # - Setting SIMUTRANS_USERDIR and SIMUTRANS_INSTALLDIR in the environment will take priority over these defaults.
+      # - Setting SIMUTRANS_USERDIR and SIMUTRANS_INSTALLDIR to empty strings will disable them.
+      # - Passing '-set_userdir' and '-set_installdir' to the game will take priority over the environment variables.
+      postBuild =
         let
-          src = fetchurl { inherit url sha256; };
+          default-home = ''$(if [ -d "$HOME/.simutrans" ]; then echo "$HOME/.simutrans"; else echo "''${XDG_DATA_HOME:-$HOME/.local/share}/simutrans"; fi)'';
         in
         ''
-          mkdir -p "$out/share/simutrans/${pakName}"
-          cd "$out/share/simutrans/${pakName}"
-          "${unzip}/bin/unzip" "${src}"
-          chmod -R +w . # some zipfiles need that
-
-          set +o pipefail # no idea why it's needed
-          toStrip=`find . -iname '*.pak' | head -n 1 | sed 's|\./\(.*\)/[^/]*$|\1|'`
-          echo "Detected path '$toStrip' to strip"
-          mv ./"$toStrip"/* .
-          rm -f "$toStrip/.directory" #pak128.german had this
-          rmdir -p "$toStrip"
-        '';
-    };
+          wrapProgram $out/bin/simutrans \
+            --run 'export SIMUTRANS_USERDIR=''${SIMUTRANS_USERDIR-${default-home}}' \
+            --run 'export SIMUTRANS_INSTALLDIR=''${SIMUTRANS_INSTALLDIR-${default-home}/paksets}' \
+            --add-flags "-set_basedir $out/share/simutrans"
 
-  /*
-    The binaries need all data in one directory; the default is directory
-     of the executable, and another option is the current directory :-/
-  */
-  withPaks =
-    paks:
-    buildEnv {
-      inherit (binaries) name;
-      paths = [ binaries ] ++ paks;
-      postBuild = ''
-        rm "$out/bin" && mkdir "$out/bin"
-        cat > "$out/bin/simutrans" <<EOF
-        #!${runtimeShell}
-        cd "$out"/share/simutrans
-        exec "${binaries}/bin/simutrans" -use_workdir "\$@"
-        EOF
-        chmod +x "$out/bin/simutrans"
-      '';
+          mkdir -p $out/share/{applications,icons/hicolor/scalable/apps}
+
+          ln -s ${src}/trunk/src/simutrans/simutrans.svg $out/share/icons/hicolor/scalable/apps/simutrans.svg
+          substitute ${src}/trunk/src/linux/simutrans.desktop $out/share/applications/simutrans.desktop \
+            --replace-fail "Exec=simutrans" "Exec=$out/bin/simutrans"
```

Comment:
We should not do this as many desktops like KDE do not properly update the paths to the binaries. So just leaving simutrans here is actually preferred.